api/src/main/java/org/openmrs/Form.java

```diff
@@ -148,8 +148,8 @@ public void setEncounterType(EncounterType encounterType) {
 				
 				for (FormField ff : fieldSet) {
 					if (ff.getFieldNumber() != null) {
-						if (ff.getFieldNumber().intValue() < fieldNum || fieldNum == 0) {
```

Comment:
Are you sure the object comparison is the same as primitive value comparison?

Reply:
It is "<" operator, so int and Integer comparisons are the same. It is different just with "==" operator.